uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/M2RepoServiceCreator.java

```diff
@@ -0,0 +1,105 @@
+import org.uberfire.mocks.MockInstanceImpl;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class M2RepoServiceCreator {
+
+    private static final Logger log = LoggerFactory.getLogger(M2MavenRepositoryServiceImplTest.class);
+    GuvnorM2Repository repo;
+    M2RepoServiceImpl service;
+    HttpPostHelper helper;
+    java.lang.reflect.Method helperMethod;
+
+    M2RepoServiceCreator() throws Exception {
+        log.info("Deleting existing Repositories instance..");
+
+        File dir = new File("repositories");
+        log.info("DELETING test repo: " + dir.getAbsolutePath());
+        deleteDir(dir);
+        log.info("TEST repo was deleted.");
+
+        ArtifactRepositoryPreference pref = mock(ArtifactRepositoryPreference.class);
+        when(pref.getGlobalM2RepoDir()).thenReturn("repositories/kie");
+        when(pref.isGlobalM2RepoDirEnabled()).thenReturn(true);
+        when(pref.isDistributionManagementM2RepoDirEnabled()).thenReturn(true);
+        when(pref.isWorkspaceM2RepoDirEnabled()).thenReturn(false);
+        WorkspaceNameResolver resolver = mock(WorkspaceNameResolver.class);
+        when(resolver.getWorkspaceName()).thenReturn("global");
+        ArtifactRepositoryProducer producer = new ArtifactRepositoryProducer(pref,
+                                                                             resolver);
+        producer.initialize();
+        Instance<ArtifactRepository> repositories = new MockInstanceImpl<>(producer.produceLocalRepository(),
+                                                                           producer.produceGlobalRepository(),
+                                                                           producer.produceDistributionManagementRepository());
+        ArtifactRepositoryService factory = new ArtifactRepositoryService(repositories);
+        repo = new GuvnorM2Repository(factory);
+        repo.init();
+
+        //Create a shell M2RepoService and set the M2Repository
+        service = new M2RepoServiceImpl(mock(Logger.class),
+                                        repo);
+        java.lang.reflect.Field repositoryField = M2RepoServiceImpl.class.getDeclaredField("repository");
+        repositoryField.setAccessible(true);
+        repositoryField.set(service,
+                            repo);
+
+        //Make private method accessible for testing
+        helper = new HttpPostHelper();
+        helperMethod = HttpPostHelper.class.getDeclaredMethod("upload",
+                                                              FormData.class);
+        helperMethod.setAccessible(true);
+
+        //Set the repository service created above in the HttpPostHelper
+        java.lang.reflect.Field m2RepoServiceField = HttpPostHelper.class.getDeclaredField("m2RepoService");
+        m2RepoServiceField.setAccessible(true);
+        m2RepoServiceField.set(helper,
+                               service);
+    }
+
+    static boolean deleteDir(File dir) {
```

Comment:
I'd recommend using something like `FileUtils.deleteDirectory(f);` from commons-io (it's already used in appformer, you can add it as test dependency to this module)